datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -229,6 +239,7 @@ protected void restartDeadTasks() {
         _logger.warn("Creating a new connector task for the datastream task {}", datastreamTask);
         _runningTasks.put(datastreamTask, createKafkaConnectorTask(datastreamTask));
       });
+      _dynamicMetricsManager.createOrUpdateCounter(MODULE, NUM_TASK_RESTARTS, deadDatastreamTasks.size());
```

Comment:
@ahmedahamid is there a recommendation on when to use a gauge vs. a counter?

Reply:
Here's what I found: https://github.com/linkedin/brooklin/pull/740#discussion_r469408157. GitHub started allowing me to leave responses on comments after I posted my own review.